server/schedulers/hot_region.go

```diff
@@ -357,23 +345,18 @@ func (h *hotScheduler) balanceHotReadRegions(cluster opt.Cluster) []*operator.Op
 	return nil
 }
 
-// balanceHotRetryLimit is the limit to retry schedule for selected balance strategy.
-const balanceHotRetryLimit = 5
-
 func (h *hotScheduler) balanceHotWriteRegions(cluster opt.Cluster) []*operator.Operator {
-	for i := 0; i < balanceHotRetryLimit; i++ {
-		// prefer to balance by peer
-		peerSolver := newBalanceSolver(h, cluster, write, movePeer)
-		ops := peerSolver.solve()
-		if len(ops) > 0 {
-			return ops
-		}
+	// prefer to balance by peer
```

Comment:
After removing the retry loop, will it affect the speed of hot spot scheduling?

Reply:
I believe it won't. The the statistics data is not changed for each iteration, and new `balanceSolver` will try almost all possible solutions.